### PR TITLE
Allows HUDs to bypass Patcher's HUD Caching

### PR DESCRIPTION
**Pull request by ImToggle**
https://github.com/Polyfrost/OneConfig/assets/98242902/6ee6d9c9-b469-4b43-bbd8-ef1c075dd8d9

## Checklist
- [x] I made a clear description of what was changed
- [x] I stated why these changes were necessary
- [x] I updated documentation or said what needs to be updated
- [x] I made sure these changes are backwards compatible
- [x] This pull request is for one feature/bug fix

### CODE REVIEW

File: src/main/java/cc/polyfrost/oneconfig/hud/Hud.java [hunk 1]
```diff
@@ -216,6 +217,13 @@ public boolean isLocked() {
         return locked && (config == null || config.enabled);
     }
 
+    /**
+     * @return If the hud is ignored from hud caching
+     */
+    public boolean isIgnored() {
+        return ignoreCaching && (config == null || config.enabled) && Platform.getLoaderPlatform().isModLoaded("patcher");
```
Comment (nextdayy) [inline src/main/java/cc/polyfrost/oneconfig/hud/Hud.java:224]: cache this value

Comment (ImToggle) [reply src/main/java/cc/polyfrost/oneconfig/hud/Hud.java:224]: the patcher check?

**Comment by ImToggle** [comment]
like that?

**Comment by ImToggle** [comment]
do it in HudCore so it only runs once

**Review by nextdayy — APPROVED** [review]

File: versions/src/main/java/cc/polyfrost/oneconfig/internal/mixin/EntityRendererMixin.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import static cc.polyfrost.oneconfig.internal.hud.HudCore.editing;
+import static cc.polyfrost.oneconfig.internal.hud.HudCore.huds;
+
+@Mixin(EntityRenderer.class)
+public class EntityRendererMixin {
+
+    @Inject(method = "updateCameraAndRender", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/GuiIngame;renderGameOverlay(F)V", shift = At.Shift.AFTER))
+    private void drawHud(float partialTicks, long nanoTime, CallbackInfo ci) {
```
Comment (nextdayy) [inline versions/src/main/java/cc/polyfrost/oneconfig/internal/mixin/EntityRendererMixin.java:44]: sorry actually why is this like this now?

Comment (ImToggle) [reply versions/src/main/java/cc/polyfrost/oneconfig/internal/mixin/EntityRendererMixin.java:44]: patcher redirects renderGameOverlay(), so i have to do it after the call, which is in this class

**Comment by ImToggle** [comment]
no it would run everytime a hud is added

File: versions/src/main/java/cc/polyfrost/oneconfig/internal/mixin/EntityRendererMixin.java (hunk 2 above)
Comment (nextdayy) [reply versions/src/main/java/cc/polyfrost/oneconfig/internal/mixin/EntityRendererMixin.java:44]: okay if it works then it works

**Review by nextdayy — DISMISSED** [review]

File: src/main/java/cc/polyfrost/oneconfig/hud/HUDUtils.java [hunk 3]
```diff
@@ -71,10 +71,12 @@ public Object setValue(Object value) {
             ArrayList<Field> fieldArrayList = ConfigUtils.getClassFields(hud.getClass());
             HashMap<String, Field> fields = new HashMap<>();
             for (Field f : fieldArrayList) fields.put(f.getName(), f);
+            boolean isPatcher = Platform.getLoaderPlatform().isModLoaded("patcher");
```
Comment (Wyvest) [inline src/main/java/cc/polyfrost/oneconfig/hud/HUDUtils.java:74]: This doesnt seem to have been updated

File: src/main/java/cc/polyfrost/oneconfig/hud/Hud.java [hunk 4]
```diff
@@ -216,6 +218,13 @@ public boolean isLocked() {
         return locked && (config == null || config.enabled);
     }
 
+    /**
+     * @return If the hud is ignored from hud caching
+     */
+    public boolean isIgnored() {
```
Comment (Wyvest) [inline src/main/java/cc/polyfrost/oneconfig/hud/Hud.java:224]: Change this to isHudCachingIgnored() because that name is too generic

**Review by Wyvest — CHANGES_REQUESTED** [review]

**Review by Wyvest — APPROVED** [review]

**Review by Wyvest — APPROVED** [review]

**Pull request merged**